smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpClientHelper.java

```diff
@@ -63,44 +69,67 @@ static AmqpClient createClientFromClientOptionsBean(Vertx vertx, Instance<AmqpCl
      *
      * @param channel the channel configuration
      * @param custom the custom configuration
+     * @param config the common configuration
      */
-    static void mergeTo(AmqpClientOptions channel, AmqpClientOptions custom) {
-        String username = channel.getUsername();
-        String password = channel.getPassword();
-        String host = channel.getHost();
-        int port = channel.getPort();
-        boolean useSsl = channel.isSsl();
-        int reconnectAttempts = channel.getReconnectAttempts();
-        long reconnectInterval = channel.getReconnectInterval();
-        int connectTimeout = channel.getConnectTimeout();
-
-        if (username != null) {
+    static void merge(AmqpClientOptions channel, AmqpClientOptions custom, AmqpConnectorCommonConfiguration config) {
```

Comment:
I don't completely follow what is going on in this method. Maybe if we move `getOptionsFromChannel` inside the `merge` method and return the merged options, it'll be easier to follow.

Reply:
done.